Sources/PointFree/Session.swift

```diff
@@ -50,11 +50,127 @@ extension URLRequest {
   }
 }
 
-public struct Session: Codable, Equatable {
+public struct Session: Equatable {
```

Comment:
the `Session` type changed quite a bit. I refactored to not let `userId` be the defining quality of the session, but rather a new enum `User = standard | ghosting`. I needed to do a decent amount of work to support backwards compatibility with the old session (luckily snapshot tests of middleware caught all regressions). I would be interested if you see any spots to simplify further.

Reply:
Yeah this was my main thought reading so far.

Reply:
Also, what the behavior of breaking? Does it just re-log you in via GitHub redirect? Is it worth letting the breakage happen if so?

Reply:
it would only re-login if you are on a page that requires authentication. otherwise you will just be silently logged out.

Reply:
K. I just finished reading and the extra code is not so bad.